src/librustc_metadata/creader.rs

```diff
@@ -853,6 +853,33 @@ impl<'a> CrateLoader<'a> {
         }
     }
 
+    fn inject_profiler_runtime(&mut self) {
+        if self.sess.opts.debugging_opts.profile {
+            let mut uses_std = false;
+            self.cstore.iter_crate_data(|_, data| {
```

Comment:
Ah ok. @whitequark perhaps this clause (checking `uses_std`) be removed for now?

Is it really feasible for someone to write their own profiler builtins?